zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -842,24 +936,32 @@ where
                         payment.amount,
                         Some(memo),
                     ));
-                } else if let Some(taddr) = ua.transparent() {
+
+                    continue;
+                }
+
+                if let Some(taddr) = ua.transparent() {
                     if payment.memo.is_some() {
                         return Err(Error::MemoForbidden);
                     } else {
                         builder.add_transparent_output(taddr, payment.amount)?;
+                        continue;
                     }
-                } else {
-                    return Err(Error::NoSupportedReceivers(
-                        ua.unknown().iter().map(|(tc, _)| *tc).collect(),
-                    ));
                 }
+
+                return Err(Error::NoSupportedReceivers(Box::new(ua.clone())));
             }
             Address::Sapling(addr) => {
```

Comment:
Maybe we could define a closure to add an output of each type, so that the Sapling and transparent code here would be deduplicated? It's not a lot of code, so non-blocking.

Reply:
This had previously been implemented using closures as you suggest, but additional changes that will be coming in #1143 make this approach a bit cleaner. I refactored it this way in #1087 in order to eventually avoid conflicts with #1143.